cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/AbstractStreamCoordinator.java

```diff
@@ -161,10 +164,13 @@ public Cancellable addListener(String streamName, StreamPropertyListener listene
   }
 
   @Override
-  public void close() throws IOException {
+  protected void shutDown() throws Exception {
     propertyStore.get().close();
+    doShutDown();
   }
 
+  protected abstract void doShutDown() throws Exception;
```

Comment:
javadoc is needed for protected abstract to state clearly the contract that sub-classes need to follow.